idc_collections/collex_metadata_utils.py

```diff
@@ -42,7 +42,8 @@ def get_bq_metadata(filters, fields, data_versions):
     """
 
     filter_attrs = Attribute.objects.filter(active=True, name__in=list(filters.keys()))
-    field_attrs = Attribute.objects.filter(active=True, name__in=fields)
+    # Preserve the order of fields parameter when building field_attrs
+    field_attrs = [Attribute.objects.get(active=True, name=field) for field in fields]
 
```

Comment:
Is there a reason we're doing this? This is performing count(fields) separate queries, which is very poor performance. If we absolutely need to preserve the order, it would be better to sort fields prior to coming in, then do this filter call with the same sort type.